Polish Astro environment variable documentation (#1562)
astronomer/docs: astro/environment-variables.md

For example, if you set `AIRFLOW__CORE__PARALLELISM` with one value in the Cloud UI and you set the same environment variable with another value in your `Dockerfile`, the value set in the Cloud UI takes precedence.

## Call environment variables in a DAG

> You can call environment variables storing your Airflow variables and connections from DAG code in the following ways: 

Reviewer: ```suggestion
You can call environment variables storing your Airflow variables and connections from DAG code in the following ways: 
```
I find this statement very confusing and I'm not sure what you're trying to convey. Maybe:

You can use the following methods to call environment variables in your DAG code:

- `Variable.get('<environment-variable-key>')`

> - `os.getenv('<environment-variable-key>', '<default_value>')`

Reviewer 2: Hm I actually think this section doesn't quite makes sense _before_ the Airflow connections and variables section

- For one, the intro is saying `storing your Airflow variables and connections` even though you haven't introduced that concept yet
- The example here is both "how to call an env var as an env var" and then "how to call variables stored as env vars"

We could split these sections up, but I'd kind of rather first have sections about _creating_ these things and then a consolidated section on calling them in your DAGs

Reviewer 2: (I agree the header shouldn't be called "query" though, that's a bit yikes)

Author: This section should serve as a more general "How to call _any_ environment variable value in a DAG", regardless of whether it's a connection or Airflow variable or something else. For that reason, I think it makes sense to include before the details about storing connections and variables. 

Rob's edits removed mentions of Airflow connections/ variables, which should make this more clear.

For non-secret values, Astronomer recommends using `os.getenv('<environment-variable-key>', '<default_value>')` to reduce the number of Airflow metadata database requests.  Replace `environment-variable-key` with the key for your environment variable on Astro and `default_value` with the default value for the key. Typically, `default_value` is the value you defined for the environment variable in the Cloud UI. When your DAG queries an environment variable that doesn't exist, `default_value` is used.

## Add Airflow connections and variables using environment variables

If you regularly use [Airflow connections](https://airflow.apache.org/docs/apache-airflow/stable/concepts/connections.html) and [variables](https://airflow.apache.org/docs/apache-airflow/stable/concepts/variables.html), Astronomer recommends storing and fetching them with environment variables instead of adding them to the Airflow UI.

Airflow connections and variables are stored in Airflow's metadata database. Calling them outside of task definitions and operators requires an additional connection to Airflow's metadata database which is used every time the scheduler parses a DAG.

> By adding connections and variables as environment variables, you can refer to them more easily in your code and lower the amount of open connections, which improves the performance of your database and resources.

Reviewer: ```suggestion
By adding connections and variables as environment variables, you can lower the amount of open connections, and improve the performance of your database and resources.
```
![image](https://user-images.githubusercontent.com/104205257/204853562-cf2a30d7-965a-4944-98c0-4cac083b4a3d.png)

### Airflow connections

> Use the Airflow [connection URI format](https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html#uri-format) to store connections as environment variables. The environment variable naming convention for Airflow connections is:

Reviewer: ```suggestion
Use the Airflow [connection URI format](https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html#uri-format) to store connections as environment variables. The naming convention for Airflow environment variable connections is:
```

- Key: `AIRFLOW_CONN_<CONN_ID>` 
- Value: `<connection-uri>`

For example, consider the following Airflow connection:

- Connection ID: `MY_PROD_DB`
- Connection URI: `my-conn-type://login:password@host:5432/schema`

To store this connection as an environment variable, you create an environment variable with the key `AIRFLOW_CONN_MY_PROD_DB` and the value `my-conn-type://login:password@host:5432/schema`,

:::info

> Airflow connections set with environment variables do not appear in the Airflow UI. They can only be seen and updated from the Cloud UI or your Dockerfile. 

Reviewer: ```suggestion
Airflow connections set with environment variables do not appear in the Airflow UI. They can only be seen and updated in the Cloud UI or your Dockerfile. 
```
